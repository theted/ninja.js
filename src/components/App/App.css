.mt-3 {
  margin-top: 2rem;
}

.mr-1 {
  margin-right: 0.5rem;
}

.pagination {
  list-style: none;
}

.pagination li {
  display: inline-block;
}
