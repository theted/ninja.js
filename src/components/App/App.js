import React from 'react';
import PropTypes from 'prop-types';
import DataTable from '../DataTable';
import './App.css';

const App = ({ rows }) => {
  return (
    <div className="container mt-3">
      <DataTable rows={rows} locale="da" rowsPerPage={5} />
    </div>
  );
};

App.propTypes = {
  rows: PropTypes.shape().isRequired,
};

export default App;
