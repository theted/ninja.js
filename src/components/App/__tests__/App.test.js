import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../App';

import users from '../../../data/users.json';

const rows = users.slice(0, 5);

const props = {
  rows,
  locale: 'da',
  rowsPerPage: 5,
};

describe('App component', () => {
  it('renders without crashing', () => {
    shallow(<App {...props} />);
  });

  it('matches snapshot', () => {
    const component = shallow(<App {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders 5 rows', () => {
    const wrapper = mount(<App rows={rows} locale="da" rowsPerPage={5} />);

    expect(wrapper.find('tr').length).toBe(5);
  });

  it('filters rows based on input', () => {
    const wrapper = mount(<App rows={rows} locale="da" rowsPerPage={5} />);

    wrapper.find('input').simulate('change', { target: { value: 'k' } });

    expect(wrapper.find('tr').length).toBe(2);
  });
});
