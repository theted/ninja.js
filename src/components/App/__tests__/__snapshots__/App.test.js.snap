// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="container mt-3"
>
  <DataTable
    locale="da"
    rows={
      Array [
        Object {
          "edit_path": "http://google.com",
          "email": "MadsLKlausen@jourrapide.com",
          "name1": "Mads L. Klausen",
          "per_id": 1,
        },
        Object {
          "edit_path": "http://google.com",
          "email": "AlfredKKrogh@armyspy.com",
          "name1": "Alfred K. Krogh",
          "per_id": 2,
        },
        Object {
          "edit_path": "http://google.com",
          "email": "SilasLBertelsen@armyspy.com",
          "name1": "Silas L. Bertelsen",
          "per_id": 3,
        },
        Object {
          "edit_path": "http://google.com",
          "email": "MiaAJohnsen@dayrep.com",
          "name1": "Mia A. Johnsen",
          "per_id": 4,
        },
        Object {
          "edit_path": "http://google.com",
          "email": "AlfredSSchou@jourrapide.com",
          "name1": "Alfred S. Schou",
          "per_id": 5,
        },
      ]
    }
    rowsPerPage={5}
  />
</div>
`;
